src/main/java/net/neoforged/neoforge/fluids/capability/wrappers/CauldronWrapper.java

```diff
@@ -0,0 +1,139 @@
+package net.neoforged.neoforge.fluids.capability.wrappers;
+
+import com.google.common.math.IntMath;
+import java.util.Objects;
+import net.minecraft.core.BlockPos;
+import net.minecraft.world.level.Level;
+import net.minecraft.world.level.block.Blocks;
+import net.minecraft.world.level.block.state.BlockState;
+import net.minecraft.world.level.material.Fluids;
+import net.neoforged.neoforge.fluids.CauldronFluidContent;
+import net.neoforged.neoforge.fluids.FluidStack;
+import net.neoforged.neoforge.fluids.capability.IFluidHandler;
+import org.jetbrains.annotations.ApiStatus;
+
+@ApiStatus.Internal
+public class CauldronWrapper implements IFluidHandler {
+    private final Level level;
+    private final BlockPos pos;
+
+    public CauldronWrapper(Level level, BlockPos pos) {
+        this.level = level;
+        this.pos = pos.immutable();
```

Comment:
If this is an actual potential problem we may want to consider doing it once and inside of `BlockCapability#getCapability`? That way impls don't have to worry about it.

Reply:
Indeed, I forgot about it for the composter...